packages/test/local-server-stress-tests/src/ddsModels.ts

```diff
@@ -0,0 +1,203 @@
+import { toFluidHandleInternal } from "@fluidframework/runtime-utils/internal";
+import {
+	baseSharedStringModel,
+	baseIntervalModel,
+	// eslint-disable-next-line import/no-internal-modules
+} from "@fluidframework/sequence/internal/test";
+
+import {
+	LocalServerStressState,
+	makeUnreachableCodePathProxy,
+	Client,
+} from "./localServerStressHarness";
+
+export function repeatFactoryAsync<T, TState = void>(
+	factory: () => AsyncGenerator<T, TState>,
+): AsyncGenerator<T, TState> {
+	let generator = factory();
+	return async (state: TState) => {
+		const next = await generator(state);
+		if (next !== done) {
+			return next;
+		}
+		generator = factory();
+		return generator(state);
+	};
+}
+
+const generateSubModelMap = (
+	...models: Omit<DDSFuzzModel<IChannelFactory, any>, "workloadName">[]
+) => {
+	const modelMap = new Map<
+		string,
+		{
+			factory: IChannelFactory;
+			generator: AsyncGenerator<any, DDSFuzzTestState<IChannelFactory>>;
+			reducer: DDSFuzzModel<IChannelFactory, any>["reducer"];
+			validateConsistency: DDSFuzzModel<IChannelFactory, any>["validateConsistency"];
+			minimizationTransforms?: DDSFuzzModel<IChannelFactory, any>["minimizationTransforms"];
+		}
+	>();
+	for (const model of models) {
+		const { reducer, generatorFactory, factory, validateConsistency, minimizationTransforms } =
+			model;
+		const generator = repeatFactoryAsync(generatorFactory);
+		modelMap.set(factory.attributes.type, {
+			generator,
+			reducer,
+			factory,
+			validateConsistency,
+			minimizationTransforms,
+		});
+	}
+
+	return modelMap;
+};
+
+export const ddsModelMap = generateSubModelMap(
+	baseMapModel,
+	baseDirModel,
+	baseSharedStringModel,
+	baseIntervalModel,
+);
+
+export interface DDSModelOp {
+	type: "DDSModelOp";
+	channelType: string;
+	channelId: string;
+	op: unknown;
+}
+
+const createDDSClient = (channel: IChannel): DDSClient<IChannelFactory> => {
+	return {
+		channel,
+		containerRuntime: makeUnreachableCodePathProxy("containerRuntime"),
+		dataStoreRuntime: makeUnreachableCodePathProxy("dataStoreRuntime"),
+	};
+};
+
+const covertLocalServerStateToDdsState = (
+	state: LocalServerStressState,
+	channel: IChannel,
+): DDSFuzzTestState<IChannelFactory> => {
+	return {
+		clients: makeUnreachableCodePathProxy("clients"),
+		client: createDDSClient(channel),
+		containerRuntimeFactory: makeUnreachableCodePathProxy("containerRuntimeFactory"),
+		isDetached: state.isDetached,
+		summarizerClient: makeUnreachableCodePathProxy("containerRuntimeFactory"),
+		random: {
+			...state.random,
+			handle: () => {
+				const realHandle = toFluidHandleInternal(
+					state.random.pick([
+						...Object.values(state.client.entryPoint.channels)
+							.flatMap((ca) => ca)
+							.map((c) => c.handle),
+						...Object.values(state.client.entryPoint.globalObjects)
+							.map((v) => v.handle)
+							.filter((v): v is IFluidHandle => v !== undefined),
+					]),
+				);
+				return {
```

Comment:
here we do some funky stuff with handles so we can serialize them like json for output, but not bind them, as they may not be attached. this doesn't currently work with minimization or replay, but i'm pretty sure we can fix it on deserialization, using the absolute path for lookup

Reply:
still funky, but now works with minimization and replay